packages/react-button/src/components/ToggleButton/useToggleButton.ts

```diff
@@ -0,0 +1,34 @@
+import * as React from 'react';
+import { mergeProps, ComposePreparedOptions } from '@fluentui/react-compose';
+import { useControllableValue } from '@uifabric/react-hooks';
+import { ToggleButtonProps } from './ToggleButton.types';
+
+/**
+ * The useToggleButton hook processes the ToggleButton component props and returns
+ * state, slots, and slotProps for consumption by the component.
+ * @param props
+ */
+export const useToggleButton = (
+  props: ToggleButtonProps,
+  options: ComposePreparedOptions,
+  parentState?: ComposePreparedOptions['state'],
+) => {
+  const { checked: controlledChecked, defaultChecked = false, onClick: onButtonClick } = props;
+  const [checked, setChecked] = useControllableValue(controlledChecked, defaultChecked);
+
+  const onClick = (ev: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+    if (onButtonClick) {
+      onButtonClick(ev);
+    }
+
+    setChecked(!checked);
+  };
+
+  const state = { ...props, checked, setChecked, onClick };
+
+  if (parentState) {
+    return parentState(state, options);
+  }
```

Comment:
I wonder if this is better:
1. having useComponentState hook which doesn't `mergeProps`
2. have Base component to `mergeProps`

Reply:
if we go what your approach, what's the advantage of this over calling `useButton` directly here?

Reply:
I guess I'll answer this in 3 parts:

- For point 1, I think even without doing this work that would be preferable.
- For point 2, did you mean calling `mergeProps` on the render function? I would prefer to not do that as it would make a hard dependency on it. I'd much rather have separate `useComponent` and `useComponentState` hooks where the latter is called by the earlier.
- Regarding the benefits of this approach over calling `useButton` directly, the whole reason behind this is to be able to substitute or add state on top of what is provided by default. You can imagine a user substituting or expanding the `useButton` state on a recompose. You could still add the toggling capabilities by using this hook on recomposition of that new component without adding anything else and calling their new state. The idea here would be to basically eliminate the hard dependencies that would be created in calling the `useButton` state.